core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -199,6 +199,23 @@ class LogTest {
   }
 
   @Test
+  def testSizeForLargeLogs(): Unit = {
+    val megaByte = 1024 * 1024
+    val log = createLog(megaByte, messagesPerSegment = 1024)
```

Comment:
Thanks for adding the test. I think this will probably use excessive resources. It seems like the easiest way to test this would be to introduce a method in the `Log` companion object: `def size(segments: Seq[LogSegment])`. And call that from `Log.size`. Then you can add a test that uses mocked `LogSegment`s. Does that make sense?